lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -97,20 +118,30 @@ def authenticate(group_name=None, internal_token_only=False):
 def authorize(group_name=None):
 
     if 'user_token_data' not in g:
-        raise OAuthProblem('Token data not available for authorization. Most likely an authentication error.')
+        raise OAuthProblem(
+            'Token data not available for authorization. Most likely an authentication error.')
     else:
         id_info = g.user_token_data
 
         if group_name is not None:
+            AUTH_ISSUER = os.getenv('AUTH_ISSUER', '').strip()
+            # So we are to check is a group membership is required.
+            # Get the membership check keys based on issuer
+            if id_info['iss'] == AUTH_ISSUER:
+                is_member_of_key = is_member_of_claim
+            else:
+                is_member_of_key = uiucedu_is_member_of
             # check if the group_name is list or string
             if isinstance(group_name, str):
                 # make group name as list
                 group_name = [group_name]
 
             is_authorize = False
-            for name in group_name:
-                if uiucedu_is_member_of in id_info:
-                    is_member_of = id_info[uiucedu_is_member_of]
+            if is_member_of_key in id_info:
+                is_member_of = id_info[is_member_of_key]
+                for name in group_name:
```

Comment:
The Core BB  tokens use comma separated lists for the permissions while Shibboleth uses a JSON list so we will need to handle this difference